clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+        unlist(use.names = FALSE)
+    dollarData <- cleanPart1 %>% filter(dollarIdx) %>%
+        mutate(ExpectedEarning = sub("\\$", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!dollarIdx) %>% bind_rows(dollarData)
+
+    # Remove commas from expected earnings
+    commaIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl(",", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    commaData <- cleanPart1 %>% filter(commaIdx) %>%
+        mutate(ExpectedEarning = sub(",", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!commaIdx) %>% bind_rows(commaData)
+
+    # Remove "k" from expected earnings
+    kIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("k", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    kData <- cleanPart1 %>% filter(kIdx) %>%
+        mutate(ExpectedEarning = sub("k", "000", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!kIdx) %>% bind_rows(kData)
+
+    # Change range of expected earnings into average
+    rangeIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("-", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    rangeData <- cleanPart1 %>% filter(rangeIdx) %>%
+        mutate(ExpectedEarning = average_range_earning(ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!rangeIdx) %>% bind_rows(rangeData)
+
+    # Remove period from salaries like 50.000 which should be just 50000
+    thousandsIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("^\\d{2}\\.", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    thousandsData <- cleanPart1 %>% filter(thousandsIdx) %>%
+        mutate(ExpectedEarning = sub("\\.", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!thousandsIdx) %>%
+        bind_rows(thousandsData)
+
+    # Remove any non-numeric characters
+    numericIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("[A-Za-z]", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    numericData <- cleanPart1 %>% filter(numericIdx) %>%
+        mutate(ExpectedEarning = gsub("[A-Za-z']*", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!numericIdx) %>%
+        bind_rows(numericData)
+
+    # Make all expected earnings less than 100 multiplied by 1000
+    below100 <- cleanPart1 %>%
+        mutate(ExpectedEarning = as.numeric(ExpectedEarning)) %>%
+        filter(ExpectedEarning < 100) %>%
+        filter(ExpectedEarning != 0) %>%
+        mutate(ExpectedEarning = as.character(ExpectedEarning))
+    change100 <- below100 %>%
+        mutate(ExpectedEarning = as.numeric(ExpectedEarning)) %>%
+        mutate(ExpectedEarning = ExpectedEarning * 1000) %>%
+        mutate(ExpectedEarning = as.character(ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% setdiff(below100) %>% bind_rows(change100)
+
```

Comment:
Also in many countries nobody use annual salaries, but monthly. Some people were probably confused and entered their monthly expectations. I also remember to be confused for a few seconds.
So it make sense to multiply by 12 all earnings between 500 and 6000